core/controllers/cron_test.py

```diff
@@ -316,6 +321,149 @@ def _mock_logging_function(msg, *args):
+            expected_reviewable_suggestion_email_info.suggestion_type)
+        self.assertEqual(
+            reviewable_suggestion_email_info.language_code,
+            expected_reviewable_suggestion_email_info.language_code)
+        self.assertEqual(
+            reviewable_suggestion_email_info.suggestion_content,
+            expected_reviewable_suggestion_email_info.suggestion_content)
+        self.assertEqual(
+            reviewable_suggestion_email_info.submission_datetime,
+            expected_reviewable_suggestion_email_info.submission_datetime)
+
+    def _mock_send_contributor_dashboard_reviewers_emails(
+            self, reviewer_ids, reviewers_suggestion_email_infos):
+        """Mocks
+        email_manager.send_mail_to_notify_contributor_dashboard_reviewers as
+        its not possible to send mail with self.testapp_swap, i.e with the URLs
+        defined in main_cron.
+        """
+        self.reviewer_ids = reviewer_ids
+        self.reviewers_suggestion_email_infos = reviewers_suggestion_email_infos
+
+    def setUp(self):
+        super(
+            CronMailContributorDashboardReviewerOpportunitiesHandlerTests,
+            self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.signup(self.AUTHOR_EMAIL, self.AUTHOR_USERNAME)
+        self.author_id = self.get_user_id_from_email(self.AUTHOR_EMAIL)
+        self.signup(self.REVIEWER_EMAIL, self.REVIEWER_USERNAME)
+        self.reviewer_id = self.get_user_id_from_email(self.REVIEWER_EMAIL)
+        user_services.update_email_preferences(
+            self.reviewer_id, True, False, False, False)
+        self.save_new_valid_exploration(self.target_id, self.author_id)
+
+        self.can_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', True)
+        self.cannot_send_emails = self.swap(feconf, 'CAN_SEND_EMAILS', False)
+        self.can_send_reviewer_emails = self.swap(
+            config_domain, 'NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_IS_ENABLED',
+            True)
+        self.cannot_send_reviewer_emails = self.swap(
+            config_domain, 'NOTIFY_CONTRIBUTOR_DASHBOARD_REVIEWERS_IS_ENABLED',
+            False)
+        self.testapp_swap = self.swap(
+            self, 'testapp', webtest.TestApp(main_cron.app))
+
+        self.reviewers_suggestion_email_infos = []
+        self.reviewer_ids = []
+
+    def test_email_not_sent_if_sending_reviewer_emails_is_not_enabled(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        with self.cannot_send_reviewer_emails, self.testapp_swap:
+            self.get_html_response(
+                '/cron/mail/contributor_dashboard_reviewers/'
+                'review_opportunities')
+
+        self.assertEqual(len(self.reviewer_ids), 0)
+        self.assertEqual(len(self.reviewers_suggestion_email_infos), 0)
```

Comment:
I'm not sure that I understand how this assertion works--these properties seem like they will never be assigned unless _mock_send_contributor_dashboard_reviewers_emails is used, which it isn't in this test. That means we could have any arrangement that we want--as long as we don't use _mock_send_contributor_dashboard_reviewers_emails this test will always pass. Or am I missing something?

Preferably, the test should be set up such that if we switched the with condition properties above, the test should fail.

Ditto for other tests here.